### DISCUSSION TITLE
Reloading custom endpoint

### DISCUSSION THREAD

**Discussion by pierre-H** [directus/directus#5539] [Archive]
First, thanks for this great tool ! Directus is great headless CMS !
When creating a custom endpoint, (with docker or `npx directus start`), editing the file needs a restart of the server to update the result.

Example : 
If I want to change
```js
// /extensions/endpoints/my-endpoint/index.js
module.exports = function registerEndpoint(router) {
	router.get('/', (req, res) => res.send('Hello world !'));
};
```
To
```js
// /extensions/endpoints/my-endpoint/index.js
module.exports = function registerEndpoint(router) {
	router.get('/', (req, res) => res.send('Hello Directus !'));
};
```
The change is taken into account only after the restarting of the server (even when the cache is disabled).

Is it possible to change that (with an env var for example) ? The development would be easier ...

**Comment by rijkvanzanten** [comment]
> Is it possible to change that (with an env var for example) ?

Not as of right now, no. Though there are a lot of discussions around extensions and their development workflow, so this is certainly something that's on the roadmap 👍🏻

**Comment by alexvdvalk** [comment]
There is a EXTENSIONS_AUTO_RELOAD environment variable which automatically reloaded extensions when they have changed.
